src/default_dict.jl

```diff
@@ -52,25 +52,16 @@ DefaultDictBase{K,V}(default::F; kwargs...) where {K,V,F} = DefaultDictBase{K,V,
 # most functions are simply delegated to the wrapped dictionary
 @delegate DefaultDictBase.d [ get, haskey, getkey, pop!,
                               start, done, next, isempty, length ]
-if isdefined(Base, :LegacyIterationCompat)
-    # resolve ambiguity
-    next(d::DefaultDictBase, state::Base.LegacyIterationCompat{I,T,S}) where {I>:DefaultDictBase,T,S} = next(d.d, state)
-    done(d::DefaultDictBase, state::Base.LegacyIterationCompat{I,T,S}) where {I>:DefaultDictBase,T,S} = done(d.d, state)
-end
+# resolve ambiguity
+next(d::DefaultDictBase, state::Base.LegacyIterationCompat{I,T,S}) where {I>:DefaultDictBase,T,S} = next(d.d, state)
+done(d::DefaultDictBase, state::Base.LegacyIterationCompat{I,T,S}) where {I>:DefaultDictBase,T,S} = done(d.d, state)
 
 # Some functions are delegated, but then need to return the main dictionary
 # NOTE: push! is not included below, because the fallback version just
 #       calls setindex!
 @delegate_return_parent DefaultDictBase.d [ delete!, empty!, setindex!, sizehint! ]
 
 similar(d::DefaultDictBase{K,V,F}) where {K,V,F} = DefaultDictBase{K,V,F}(d.default; passkey=d.passkey)
-if isdefined(Base, :KeySet) # 0.7.0-DEV.2722
-    in(key, v::Base.KeySet{K,T}) where {K,T<:DefaultDictBase{K}} = key in keys(v.dict.d)
-    next(v::Base.KeySet{K,T}, i) where {K,T<:DefaultDictBase{K}} = (v.dict.d.keys[i], Base.skip_deleted(v.dict.d,i+1))
-else
-    in(key, v::Base.KeyIterator{T}) where {T<:DefaultDictBase} = key in keys(v.dict.d)
-    next(v::Base.KeyIterator{T}, i) where {T<:DefaultDictBase} = (v.dict.d.keys[i], Base.skip_deleted(v.dict.d,i+1))
-end
```

Comment:
I do not believe you intended to delete both the `if` clause and the `else` clause
I think you intended to leave the `KeySet` code here.
And just get rid of `KeyIterator`

Or perhaps not, since 
https://github.com/JuliaCollections/DataStructures.jl/pull/393/files#diff-69860e53de6ca0b86046a70a1dd966cbR152
defines it for things via the meta-programming loop?

Reply:
It wasn't needed so thats why I removed it... Perhaps it will hit a fallback and this is an optimization? I can add it back, but would be good to know the reason for them. Perhaps they are just not tested?

Reply:
I wonder if we can delete the other ones instead?
I would much rather be hitting code defined on an abstract type,
than be generating methods via an eval loop for all subtypes of that abstract type.

Inheritance > Metaprogramming